cfgmgr/buffermgr.cpp

```diff
@@ -153,94 +152,99 @@ task_process_status BufferMgr::doSpeedUpdateTask(string port, bool admin_up)
+            return task_process_status::task_invalid_entry;
         }
 
-        // profile threshold field name
-        mode += "_th";
+        // check if profile already exists - if yes - skip creation
+        m_cfgBufferProfileTable.get(buffer_profile_key, fvVectorProfile);
+        // Create record in BUFFER_PROFILE table
+        if (fvVectorProfile.size() == 0)
+        {
+            SWSS_LOG_NOTICE("Creating new profile '%s'", buffer_profile_key.c_str());
+
+            string mode = getPgPoolMode();
+            if (mode.empty())
+            {
+                // this should never happen if switch initialized properly
+                SWSS_LOG_INFO("PG lossless pool is not yet created");
+                return task_process_status::task_need_retry;
+            }
+
+            // profile threshold field name
+            mode += "_th";
 
-        fvVectorProfile.push_back(make_pair("pool", INGRESS_LOSSLESS_PG_POOL_NAME));
-        fvVectorProfile.push_back(make_pair("xon", m_pgProfileLookup[speed][cable].xon));
-        if (m_pgProfileLookup[speed][cable].xon_offset.length() > 0) {
-            fvVectorProfile.push_back(make_pair("xon_offset",
-                                         m_pgProfileLookup[speed][cable].xon_offset));
+            fvVectorProfile.push_back(make_pair("pool", INGRESS_LOSSLESS_PG_POOL_NAME));
+            fvVectorProfile.push_back(make_pair("xon", m_pgProfileLookup[speed][cable].xon));
+            if (m_pgProfileLookup[speed][cable].xon_offset.length() > 0) {
+                fvVectorProfile.push_back(make_pair("xon_offset",
+                                            m_pgProfileLookup[speed][cable].xon_offset));
+            }
+            fvVectorProfile.push_back(make_pair("xoff", m_pgProfileLookup[speed][cable].xoff));
+            fvVectorProfile.push_back(make_pair("size", m_pgProfileLookup[speed][cable].size));
+            fvVectorProfile.push_back(make_pair(mode, m_pgProfileLookup[speed][cable].threshold));
+            m_cfgBufferProfileTable.set(buffer_profile_key, fvVectorProfile);
+        }
+        else
+        {
+            SWSS_LOG_NOTICE("Reusing existing profile '%s'", buffer_profile_key.c_str());
         }
-        fvVectorProfile.push_back(make_pair("xoff", m_pgProfileLookup[speed][cable].xoff));
-        fvVectorProfile.push_back(make_pair("size", m_pgProfileLookup[speed][cable].size));
-        fvVectorProfile.push_back(make_pair(mode, m_pgProfileLookup[speed][cable].threshold));
-        m_cfgBufferProfileTable.set(buffer_profile_key, fvVectorProfile);
-    }
-    else
-    {
-        SWSS_LOG_NOTICE("Reusing existing profile '%s'", buffer_profile_key.c_str());
-    }
 
-    /* Check if PG Mapping is already then log message and return. */
-    for (auto& prop : fvVectorPg)
-    {
-        if ((fvField(prop) == "profile") && (profile_ref == fvValue(prop)))
+        /* Check if PG Mapping is already then log message and return. */
+        for (auto& prop : fvVectorPg)
```

Comment:
with the changes suggested above, loop through the pgs again and retrieve the buffer pg table values for each lossless pg

Reply:
Updated. Thanks